kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/strings.kt

```diff
@@ -0,0 +1,23 @@
+package io.kotest.property.arbitrary
+
+import com.mifmif.common.regex.Generex
+import io.kotest.property.RandomSource
+import io.kotest.property.Sample
+import kotlin.random.asJavaRandom
+import kotlin.random.nextInt
+
+/**
+ * Generate strings that match the given pattern.
+ *
+ * The returned arb uses the [Generex](https://github.com/mifmif/Generex) library to generate strings. Generex
+ * supports a very restricted subset of regular expression constructs.
+ */
+fun Arb.Companion.stringPattern(pattern: String) = object : Arb<String> {
```

Comment:
I can rename this back to regex if desired. I think `regex` is very misleading since the regular expression language support is very restricted and the caller does not get any indication if any unsupported regex structure is in the provided pattern. This certainly confused me up when I tried using it.

I could not find any documentation of what is supported by Generex or I would link to that in the doc.